akka-actor-tests/src/test/scala/akka/util/AsciiStringCopySpec.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util
+import org.scalatest.Matchers
+import org.scalatest.WordSpec
+
+class AsciiStringCopySpec extends WordSpec with Matchers {
+
+  "The copyUSAsciiStrToBytes optimization" must {
+
+    "copy string internal representation successfully" in {
+      val ascii = "abcdefghijklmnopqrstuvxyz"
+      val byteArray = new Array[Byte](ascii.length)
+      Unsafe.copyUSAsciiStrToBytes(ascii, byteArray)
+      new String(byteArray) should ===(ascii)
```

Comment:
that constructor is deprecated, charset is needed

or use the same optimized way as we use in Artery: `EnvelopeBuffer.readLiteral`